### PR TITLE
Added Order, User, ShoppingCart controllers

### PR DESCRIPTION
**Pull request by aleksandr-hrankin**

### CODE REVIEW

**Review by IrynaDim — APPROVED** [review]

File: src/main/java/ua/antibyte/cinema/controller/Inject.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        CinemaHall whiteRoom = new CinemaHall(100, "white");
+        cinemaHallService.add(whiteRoom);
+
+        MovieSession movieSession = createMovieSession(ironMan, whiteRoom, LocalDateTime.now());
+        movieSessionService.add(movieSession);
+
+        authenticationService.register("bob@gmail.com", "1234");
+
+        return "inject";
```
Comment (KseniiaMakarova) [inline src/main/java/ua/antibyte/cinema/controller/Inject.java]:
```suggestion
        return "data was successfully injected";
```

File: src/main/java/ua/antibyte/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @PostMapping("/complete")
+    public void complete(@RequestParam Long userId) {
+        User user = userService.findById(userId);
+        List<Ticket> tickets = shoppingCartService.getByUser(user).getTickets();
+        orderService.completeOrder(tickets, user);
+    }
+
+    @GetMapping("/by-userId")
```
Comment (KseniiaMakarova) [inline src/main/java/ua/antibyte/cinema/controller/OrderController.java]:
```suggestion
    @GetMapping
```

File: src/main/java/ua/antibyte/cinema/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import ua.antibyte.cinema.model.dto.ShoppingCartResponseDto;
+import ua.antibyte.cinema.service.MovieSessionService;
+import ua.antibyte.cinema.service.ShoppingCartService;
+import ua.antibyte.cinema.service.UserService;
+import ua.antibyte.cinema.service.mapper.ShoppingCartMapper;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartMapper shoppingCartMapping;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/antibyte/cinema/controller/ShoppingCartController.java]:
```suggestion
    private final ShoppingCartMapper shoppingCartMapper;
```

File: src/main/java/ua/antibyte/cinema/model/dto/ShoppingCartResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package ua.antibyte.cinema.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/antibyte/cinema/model/dto/ShoppingCartResponseDto.java]: aren't id and userId the same?

File: src/main/java/ua/antibyte/cinema/service/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -22,13 +21,10 @@ public AuthenticationServiceImpl(UserService userService,
...
     public User login(String email, String password)
             throws AuthenticationException {
-        Optional<User> optionalUser = userService.findByEmail(email);
-        if (optionalUser.isPresent()) {
-            User user = optionalUser.get();
-            String hashPassword = HashUtil.hashPassword(password, user.getSalt());
-            if (user.getPassword().equals(hashPassword)) {
-                return user;
-            }
+        User user = userService.findByEmail(email);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/antibyte/cinema/service/security/AuthenticationServiceImpl.java]: the problem is that previously, if user entered incorrect email, the AuthenticationException was thrown, and now - NoSuchElementException

I would suggest rather to call optional.get() in UserController
what do you think?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**